dfmux/src/DfMuxCollector.cxx

```diff
@@ -96,6 +97,10 @@ RawTimestampToTimeCode(RawTimestamp stamp)
 				stamp.y = last_stamp.y;
 		}
 	}
+
+	// fix the subsecond counter if the iceboard is using
+	// an internal clock with a rate other than 100 MHz
+	stamp.ss = htole32(le32toh(stamp.ss) * clock_scale);
```

Comment:
Does this work if the clock is *faster* than 100 MHz? Won't it introduce round-off errors?

Reply:
Empirically it seems to work for 125 MHz clocks.  We should probably do a more thorough check, and/or avoid the round-trip endianness conversion.

Reply:
It's not the endianness that worries me, it's the integer data type. Time deltas of 8 ns (1/125 MHz) are not representable with an integer data type and 3G units. I can't imagine +/- 10 ns jitter is meaningful for your application -- and it wouldn't be at all surprising if the "synchronized" sampling is only good to a few ns to begin with -- but I wanted to flag it here.

Reply:
I'll use doubles internally and convert to integer only on return